### DISCUSSION TITLE
dwd-mosmix, --belchertown UND --database

### DISCUSSION THREAD

**Discussion by hoetzgit** [roe-dl/weewx-DWD#21] [Ideas]
Hallo, 

ich habe mal folgende Anpassungen gemacht, wenn ich `--belchertown` und `--database` verwende. Ich möchte gerne den Zeitrahmen des Meteogamms analog zur Belchertown Vorhersage haben. Über die `graphs.conf` habe ich das nicht hinbekommen.

Also schreibe ich bei `--belchertown` und `--database` in `write_database` nur Vorhersagewerte in die Datenbank, die `>=` der aktuellen Stunde und `<=` aktueller Tag + 7 "00:00" sind. Dann ist der Zeitraum gleich zur angezeigten Belchertown Vorhersage.

Macht es Sinn das evtl. einzubauen? 

Viele Grüße
Henry

**Comment by roe-dl** [comment]
Ab der aktuellen Stunde zu beginnen, ist auf jeden Fall sinnvoll, und das beißt sich auch nicht mit etwas anderem. `--html --hourly` beginnt auch bei der aktuellen Stunde.

Bei der Länge sieht es anders aus. Hier gibt es nicht nur `--belchertown` sondern auch `--html --daily`. Letzteres umfaßt die vollen 11 Tage, und da wünscht man sich dann, daß es auch `--database` tut. Da könnte man höchstens eine Option vorsehen, wie lang der Zeitraum sein soll. Es genügt auch nicht, nachzusehen, welche von den Optionen noch aktiv sind, weil man auch komplett alle Optionen in einem Aufruf angeben kann, um alle Ausgaben auf einmal zu bekommen.

Beispiel [`--html --daily --hourly --database`](https://www.woellsdorf-wetter.de/forecast/doebeln.html)
 
Wie wäre es mit einem Pull-Request?

**Reply by hoetzgit** [reply]
Für einen PR ist es noch zu früh, Deine Bemerkungen sind ja absolut richtig - die Idee muss noch etwas reifen. Mir wäre es auch lieber, dass ganze nachgelagert zu machen, hat man denn Deiner Meinung nach eine Chance das in der `graphs.conf` für das Meteogramm zu machen?

**Reply by roe-dl** [reply]
Ich habe noch keinen Weg gefunden. Wenn man bei `time_length` den Wert 604800 angibt, nimmt er leider die Woche vom letzten Tag rückwärts und nicht vom ersten Tag vorwärts.
